### DISCUSSION TITLE
Add custom log formatters as extensions

### DISCUSSION THREAD

**Discussion by enzolupia** [directus/directus#13317] [Feature Requests]
Add the ability to create custom log formatters for Pino instances by means of custom extensions.

The same workflow and lifecycle used for creating custom interfaces, displays, hooks could be extended to also allow creating logger formatters. 
The logger.ts script should then load the custom formatters and append them to the Pino instance's options.

The extension's options could include:

- formatters
- additional metadata list

**Comment by azrikahar** [comment]
It is currently possible to pass options to the Pino instance via `LOGGER_*` environment variables (link to docs: https://docs.directus.io/configuration/config-options/#general, the info banner below the table). In case you're aware of it and prefer customizing it via extensions, is there any constraints/limits in particular with the current approach?

**Reply by enzolupia** [reply]
Well, I actually wasn't able to define a custom JS formatter by using environment variables, I suppose that the envs parsing code may not work as expected when passing arbitrary JS code instead of static values. Is there any guideline to do that? Still this would be a quite difficult solution to implement, expecially when the formatter code is more complex.

**Comment by rijkvanzanten** [comment]
You should also be able to use custom pino formatters by setting `LOG_STYLE` to `raw` and piping Directus' output to a formatter, like `directus start | pino-colada`. I think I prefer that approach over having a full extension type (with all the overhead that comes with that) for just log formatters 🙂

**Reply by enzolupia** [reply]
@rijkvanzanten yes, that's actually the solution we switched to at the moment.
I understand that my proposal requires a lot of effort, so that's ok, we will stick with the standard Pino transports.
Thanks!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
